src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -57,7 +58,10 @@ public function process(ContainerBuilder $container): void
                 $invokableRef = null;
             }
 
-            $aliases = $tags[0]['command'] ?? str_replace('%', '%%', $class::getDefaultName() ?? '');
+            /** @var AsCommand|null $attribute */
+            $attribute = ($r->getAttributes(AsCommand::class)[0] ?? null)?->newInstance();
+
+            $aliases = str_replace('%', '%%', $tags[0]['command'] ?? $attribute?->name ?? '');
```

Comment:
We still need to call the deprecated method if the attribute is not present, don't we?

Reply:
on it, thanks for pointing it out!